org.lflang/src/org/lflang/generator/cpp/CppExtensions.kt

```diff
@@ -0,0 +1,305 @@
+package org.lflang.generator.cpp
+
+import org.eclipse.emf.ecore.resource.Resource
+import org.lflang.*
+import org.lflang.lf.*
+import java.nio.file.Path
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+
+/*************
+ * Copyright (c) 2019-2021, TU Dresden.
+
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted provided that the following conditions are met:
+
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+ * THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ ***************/
+
+/* *******************************************************************************************
+ * The following definitions are shortcuts to access static members of FileConfig and ASTUtils
+ *
+ * TODO these should likely be moved to a common place in the future
+ */
+
+val Resource.name: String get() = FileConfig.getName(this)
+
+fun Path.toUnixString(): String = FileConfig.toUnixString(this)
+fun Path.createDirectories() = FileConfig.createDirectories(this)
```

Comment:
Actually kotlin 1.5.0 added an identical extension to the standard library. I think I need some more config to make it available, as that part of the stdlib is JVM-specific and distributed in a separate dependency.